libsel4/arch_include/x86/interfaces/sel4arch.xml

```diff
@@ -105,6 +205,23 @@
                 description='The IOSpace to map the page table into.'/>
             <param dir="in" name="ioaddr" type="seL4_Word"
                 description='The address to map the page table at.'/>
+            <error name="seL4_DeleteFirst">
+                <description>
+                    All required page tables are already mapped in <texttt text="iospace"/> at <texttt text="ioaddr."/>
+                </description>
+            </error>
+            <error name="seL4_IllegalOperation">
+                <description>
+                    The <texttt text="_service"/> is a CPtr to a capability of the wrong type.
+                </description>
+            </error>
+            <error name="seL4_InvalidCapability">
+                <description>
+                    The <texttt text="_service"/> or <texttt text="iospace"/> is a CPtr to a capability of the wrong type.
+                    Or, <texttt text="iospace"/> is not assigned to a PCI device.
+                    Or, <texttt text="_service"/> is already mapped in an IOSpace.
+                </description>
+            </error>
```

Comment:
I'm confused about how this could happen. Since there is only one IO PageTable object type, wouldn't mapping an IO PageTable always fill in the first open slot that is missing in the hieararchy?

Reply:
I need to look at the code again, but what I thought when I wrote this, I think, was that the lookup could hit a page instead of an empty slot. I might well be wrong about that.

Reply:
I didn't think that would be possible since it only allows 4K pages to be mapped.

Here's the code for reference:
https://github.com/seL4/seL4/blob/12.1.0/src/arch/x86/object/iospace.c#L258

It looks like it's checking for the case where the page tables are full separately.